pkg/restapi/webhook/webhook.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+const notificationInterval = 2 * time.Second
+const notificationSendTimeout = 2 * time.Second
+
+var logger = log.New("aries-framework/agentd")
+
+type dispatcher struct {
+	client         *http.Client
+	subscriberURLs []string
+}
+
+// StartWebhookDispatcher will launch the webhook dispatcher in a new goroutine.
+// The dispatcher is responsible for sending notifications to subscribers.
+func StartWebhookDispatcher(subscriberURLs []string) {
+	if subscriberURLs != nil {
+		dispatcher := &dispatcher{
+			client:         &http.Client{},
+			subscriberURLs: subscriberURLs,
+		}
+
+		go dispatcher.start()
+	} else {
+		logger.Warnf("No subscriber URLs provided. Webhook dispatcher will not start.")
+	}
+}
+
+func (d *dispatcher) start() {
+	// TODO: Data should should be pushed whenever a record is created or its state property is updated  (see #472).
+	ticker := time.NewTicker(notificationInterval)
+	defer ticker.Stop()
+	for range ticker.C {
+		d.dispatch()
+	}
+}
+
+func (d *dispatcher) dispatch() {
+	for _, subscriberURL := range d.subscriberURLs {
+		err := sendNotification(subscriberURL, d.client)
+		if err != nil {
+			logger.Errorf(err.Error())
+		}
+	}
+}
+
+func sendNotification(destination string, c *http.Client) error {
+	//TODO: This HTTP post request needs to supply real data, not this sample text (see #472).
+	ctx, cancel := context.WithTimeout(context.Background(), notificationSendTimeout)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, "POST", destination,
+		bytes.NewBufferString("Sample notification from aries-agentd."))
+	if err != nil {
+		return fmt.Errorf("failed to create new http post request for %s: %s", destination, err)
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to post notification to %s: %s", destination, err)
+	}
+
+	defer func() {
+		e := resp.Body.Close()
```

Comment:
it is awful I mean `e` instead of `err` you can create a helper function
```
func closeOrLog(c io.Closer) {
	if err := c.Close(); err != nil {
		logger.Errorf("Failed to close response body")
	}
}
```

Reply:
Done!